rust/relay/src/auth.rs

```diff
@@ -0,0 +1,193 @@
+use base64::prelude::BASE64_STANDARD_NO_PAD;
+use base64::Engine;
+use once_cell::sync::Lazy;
+use sha2::digest::FixedOutput;
+use sha2::Sha256;
+use std::borrow::ToOwned;
+use std::time::{Duration, SystemTime};
+use stun_codec::rfc5389::attributes::{MessageIntegrity, Realm, Username};
+
+// TODO: Upstream a const constructor to `stun-codec`.
+pub static FIREZONE: Lazy<Realm> = Lazy::new(|| Realm::new("firezone".to_owned()).unwrap());
```

Comment:
Can we use the very recently stabilized [`OnceCell`](https://doc.rust-lang.org/std/cell/struct.OnceCell.html) so we don't need the `once_cell` crate?

Reply:
Not directly. It doesn't have a ctor that takes the closure.

Really, the upstream lib should have a const ctor for `Realm` here IMO.

Reply:
Not possible with the current API exposed by std.